traffic_ops/testing/api/v14/withobjs.go

```diff
@@ -0,0 +1,93 @@
+// All tests in the Traffic Ops API Testing framework use the same Traffic Ops instance, with persistent data. Because of this, when any test fails, all subsequent tests should be considered invalid, irrespective whether they pass or fail. Users are encouraged to use `go test -failfast`.
+func WithObjs(t *testing.T, objs []TCObj, f func()) {
+	for _, obj := range objs {
+		defer withFuncs[obj].Delete(t)
+		withFuncs[obj].Create(t)
+	}
+	f()
+}
+
+type TCObj int
+
+const (
+	CacheGroups TCObj = iota
+	CDNs
+	CDNFederations
+	Coordinates
+	DeliveryServices
+	DeliveryServiceRequests
+	DeliveryServiceRequestComments
+	Divisions
+	Origins
+	Parameters
+	PhysLocations
+	Profiles
+	ProfileParameters
+	Regions
+	Roles
+	Servers
+	Statuses
+	StaticDNSEntries
+	SteeringTargets
+	Tenants
+	Types
+	Users
+	UsersDeliveryServices
+)
+
+type TCObjFuncs struct {
+	Create func(t *testing.T)
+	Delete func(t *testing.T)
+}
+
+var withFuncs = map[TCObj]TCObjFuncs{
+	CacheGroups:                    {CreateTestCacheGroups, DeleteTestCacheGroups},
+	CDNs:                           {CreateTestCDNs, DeleteTestCDNs},
+	CDNFederations:                 {CreateTestCDNFederations, DeleteTestCDNFederations},
+	Coordinates:                    {CreateTestCoordinates, DeleteTestCoordinates},
+	DeliveryServices:               {CreateTestDeliveryServices, DeleteTestDeliveryServices},
+	DeliveryServiceRequests:        {CreateTestDeliveryServiceRequests, DeleteTestDeliveryServiceRequests},
+	DeliveryServiceRequestComments: {CreateTestDeliveryServiceRequestComments, DeleteTestDeliveryServiceRequestComments},
+	Divisions:                      {CreateTestDivisions, DeleteTestDivisions},
+	Origins:                        {CreateTestOrigins, DeleteTestOrigins},
+	Parameters:                     {CreateTestParameters, DeleteTestParameters},
+	PhysLocations:                  {CreateTestPhysLocations, DeleteTestPhysLocations},
+	Profiles:                       {CreateTestProfiles, DeleteTestProfiles},
+	ProfileParameters:              {CreateTestProfileParameters, DeleteTestProfileParameters},
+	Regions:                        {CreateTestRegions, DeleteTestRegions},
+	Roles:                          {CreateTestRoles, DeleteTestRoles},
+	Servers:                        {CreateTestServers, DeleteTestServers},
+	Statuses:                       {CreateTestStatuses, DeleteTestStatuses},
+	StaticDNSEntries:               {CreateTestStaticDNSEntries, DeleteTestStaticDNSEntries},
+	SteeringTargets:                {CreateTestSteeringTargets, DeleteTestSteeringTargets},
+	Tenants:                        {CreateTestTenants, DeleteTestTenants},
+	Types:                          {CreateTestTypes, DeleteTestTypes},
+	Users:                          {CreateTestUsers, ForceDeleteTestUsers},
+	UsersDeliveryServices: {CreateTestUsersDeliveryServices, DeleteTestUsersDeliveryServices},
```

Comment:
minor nit -- indentation that `go fmt` would fix..

Reply:
`gofmt` did that, looks like a quirk/bug of `gofmt`

Reply:
hmm..   fixed it for me running go 1.11.1